OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -48,6 +48,17 @@ public override Activity Tick(Actor self)
 
 			if (IsCanceling || target.Type == TargetType.Invalid)
 			{
+				// Make aircraft climb back to CruiseAltitude before moving on to NextActivity or going idle.
+				// TODO: Try to fix that even if NextActivity is not null, VTOLs will not turn towards the target while climbing,
+				// as NextActivity's target is not known to this activity.
+				// TODO2: Handle the LandOnIdle case if NextActivity == null.
```

Comment:
This is fairly straightforward:
```csharp
if (IsCanceling || target.Type == TargetType.Invalid)
{
	// We must return the actor to a sensible height before continuing
	// If the aircraft lands when idle and is idle, continue landing
	// Otherwise climb back to CruiseAltitude
	// TODO: Allow specific NextActivities to opt out of this height adjustment
	var continueLanding = aircraft.Info.LandWhenIdle && self.CurrentActivity.IsCanceling && self.CurrentActivity.NextActivity == null;

	if (!continueLanding)
	{
		var dat = self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition);
		if (dat > aircraft.LandAltitude && dat < aircraft.Info.CruiseAltitude)
		{
			QueueChild(self, new TakeOff(self), true);
			return this;
		}

		aircraft.RemoveInfluence();
		return NextActivity;
	}
}
```

We'll also want something similar (but to land) in `TakeOff`, but can be left to a followup PR.